src/gmp_tls_certificates.c

```diff
@@ -205,3 +205,223 @@ get_tls_certificates_run (gmp_parser_t *gmp_parser, GError **error)
+  tls_certificate_t new_tls_certificate;
+
+  entity = (entity_t) create_tls_certificate_data.context->first->data;
+
+  copy = entity_child (entity, "copy");
+
+  if (copy)
+    {
+      /* Copy from an existing tls_certificate and exit. */
+
+      name = entity_child (entity, "name");
+      comment = entity_child (entity, "comment");
+      switch (copy_tls_certificate (name ? entity_text (name) : NULL,
+                                    comment ? entity_text (comment) : NULL,
+                                    entity_text (copy), &new_tls_certificate))
+        {
+          case 0:
+            {
+              char *uuid;
+              uuid = tls_certificate_uuid (new_tls_certificate);
+              SENDF_TO_CLIENT_OR_FAIL (XML_OK_CREATED_ID
+                                        ("create_tls_certificate"),
+                                       uuid);
+              log_event ("tls_certificate", "TLS Certificate", uuid, "created");
+              free (uuid);
+              break;
+            }
+          case 1:
+            SEND_TO_CLIENT_OR_FAIL
+             (XML_ERROR_SYNTAX ("create_tls_certificate",
+                                "TLS Certificate exists already"));
+            log_event_fail ("tls_certificate", "TLS Certificate", NULL,
+                            "created");
+            break;
+          case 2:
+            if (send_find_error_to_client ("create_tls_certificate",
+                                           "tls_certificate",
+                                           entity_text (copy),
+                                           gmp_parser))
+              {
+                error_send_to_client (error);
+                return;
+              }
+            log_event_fail ("tls_certificate",
+                            "TLS Certificate",
+                            NULL,
+                            "created");
+            break;
+          case 99:
+            SEND_TO_CLIENT_OR_FAIL
+             (XML_ERROR_SYNTAX ("create_tls_certificate",
+                                "Permission denied"));
+            log_event_fail ("tls_certificate",
+                            "TLS Certificate",
+                            NULL,
+                            "created");
+            break;
+          case -1:
+          default:
+            SEND_TO_CLIENT_OR_FAIL
+             (XML_INTERNAL_ERROR ("create_tls_certificate"));
+            log_event_fail ("tls_certificate",
+                            "TLS Certificate",
+                            NULL,
+                            "created");
+            break;
+        }
+      create_tls_certificate_reset ();
+      return;
+    }
+
+  /* Check given info. */
+
+  name = entity_child (entity, "name");
+  comment = entity_child (entity, "comment");
+  certificate = entity_child (entity, "certificate");
+
+  switch (create_tls_certificate
+                (name ? entity_text (comment) : NULL,
```

Comment:
should be entity_text (name), yes?

Reply:
Correct, I've changed it in 106673d51b67f0c62e005f3a64236f27b81d4586.